### PR TITLE
Mmichaelis/updating tables

### PR DESCRIPTION
**Pull request by A-Tanner**
Adds code listings and tests for Table 13.01

Fixes [#1529](https://github.com/IntelliTect-dev/EssentialCSharp.Tooling/issues/1529)

### CODE REVIEW

File: src/Chapter13/Chapter13.csproj [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <ChapterNumber>13</ChapterNumber>
+    <LangVersion>10.0</LangVersion>
```
Comment (Keboo) [inline src/Chapter13/Chapter13.csproj]: IS there a reason we are setting this at the project level? We have a Directory.Build.props file that has a single version that applies to all projects.

File: src/Chapter13.Tests/Table13.01.Tests.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter13.Table13_01.Tests;
+
+[TestClass]
+public class LambdaHighlightTests
+{
+    public TestContext TestContext { get; set; } = null!; // Set by MSTest;
+
+    [TestMethod]
+    /* 1. */ [DataRow(".a", "CS0023")]
```
Comment (Keboo) [inline src/Chapter13.Tests/Table13.01.Tests.cs]: nicely done.

**Review by Keboo — COMMENTED** [review]
It looks like these are different from the latest changes in the document let me know when ready to review.

File: src/Chapter13.Tests/Table13.01.Tests.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+using AddisonWesley.Michaelis.EssentialCSharp.Shared.Tests;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter13.Table13_01.Tests;
+
+[TestClass]
+public class LambdaHighlightTests
+{
+    public TestContext TestContext { get; set; } = null!; // Set by MSTest;
```
Comment (Keboo) [inline src/Chapter13.Tests/Table13.01.Tests.cs]: nit: appears to be un-used
```suggestion
```

File: src/Chapter13/Table13.01.a.LambdaExpressionNotesAndExamples.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter13.Table13_01;
+
+public partial class LambdaExpressionNotesAndExamples
+{
+    // 1.
+    static public void DiscardParameters()
+    {
+#if !NET6_0_OR_GREATER
+        Action<int, int> x = (_,_)=>
```
Comment (Keboo) [inline src/Chapter13/Table13.01.a.LambdaExpressionNotesAndExamples.cs]: nit: We generally add a space after commas
```suggestion
        Action<int, int> x = (_, _)=>
```

File: src/Chapter13/Table13.01.d.LambdaExpressionNotesAndExamples.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
+﻿// Justification: Only snippets of source code shown for elucidation.
+#pragma warning disable CS0168 // Variable is declared but never used
+
+namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter13.Table13_01;
+
+public partial class LambdaExpressionNotesAndExamples
+{
+    // 4.
+    static public void MemberMethodsOnExpressions()
```
Comment (Keboo) [inline src/Chapter13/Table13.01.d.LambdaExpressionNotesAndExamples.cs]:
```suggestion
    public static void MemberMethodsOnExpressions()
```

File: src/Chapter13/Table13.01.e.LambdaExpressionNotesAndExamples.cs [hunk 6]
```diff
@@ -0,0 +1,20 @@
+﻿// Justification: Only snippets of source code shown for elucidation.
+#pragma warning disable CS0168 // Variable is declared but never used
+
+namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter13.Table13_01;
+
+public partial class LambdaExpressionNotesAndExamples
+{
+    // 5.
+    static public void PatternMatchingOnType()
```
Comment (Keboo) [inline src/Chapter13/Table13.01.e.LambdaExpressionNotesAndExamples.cs]:
```suggestion
    public static void PatternMatchingOnType()
```

File: src/Chapter13/Table13.01.a.LambdaExpressionNotesAndExamples.cs [hunk 7]
```diff
@@ -0,0 +1,16 @@
+﻿// Justification: Only snippets of source code shown for elucidation.
+#pragma warning disable CS0168 // Variable is declared but never used
+
+namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter13.Table13_01;
+
+public partial class LambdaExpressionNotesAndExamples
+{
+    // 1.
+    static public void DiscardParameters()
```
Comment (Keboo) [inline src/Chapter13/Table13.01.a.LambdaExpressionNotesAndExamples.cs]:
```suggestion
    public static void DiscardParameters()
```

File: src/Chapter13/Table13.01.f.LambdaExpressionNotesAndExamples.cs [hunk 8]
```diff
@@ -0,0 +1,19 @@
+﻿// Justification: Only snippets of source code shown for elucidation.
+#pragma warning disable CS0168 // Variable is declared but never used
+
+namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter13.Table13_01;
+
+public partial class LambdaExpressionNotesAndExamples
+{
+    // 6.
+    static public void ConvertingToImproperDelegate()
```
Comment (Keboo) [inline src/Chapter13/Table13.01.f.LambdaExpressionNotesAndExamples.cs]:
```suggestion
    public static void ConvertingToImproperDelegate()
```

File: src/Chapter13/Table13.01.d.LambdaExpressionNotesAndExamples.cs [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    {
+    //#if COMPILEERROR
```
Comment (Keboo) [inline src/Chapter13/Table13.01.d.LambdaExpressionNotesAndExamples.cs:11]: Do we need these commented lines?

File: src/Chapter13/Table13.01.g.LambdaExpressionNotesAndExamples.cs [hunk 10]
```diff
@@ -0,0 +1,30 @@
+﻿// Justification: Only snippets of source code shown for elucidation.
+#pragma warning disable CS0168 // Variable is declared but never used
+
+namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter13.Table13_01;
+
+public partial class LambdaExpressionNotesAndExamples
+{
+    // 7.
+    static public void JumpStatementsToOutOfScopeDestinations()
```
Comment (Keboo) [inline src/Chapter13/Table13.01.g.LambdaExpressionNotesAndExamples.cs]:
```suggestion
    public static void JumpStatementsToOutOfScopeDestinations()
```

File: src/Chapter13/Table13.01.h.LambdaExpressionNotesAndExamples.cs [hunk 11]
```diff
@@ -0,0 +1,21 @@
+﻿// Justification: Only snippets of source code shown for elucidation.
+#pragma warning disable CS0168 // Variable is declared but never used
+
+namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter13.Table13_01;
+
+public partial class LambdaExpressionNotesAndExamples
+{
+    // 8.
+    static public void AccessingParametersAndLocalsOutOfBody()
```
Comment (Keboo) [inline src/Chapter13/Table13.01.h.LambdaExpressionNotesAndExamples.cs]:
```suggestion
    public static void AccessingParametersAndLocalsOutOfBody()
```

File: src/Chapter13/Table13.01.g.LambdaExpressionNotesAndExamples.cs [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+    {
+//#if COMPILEERROR
+#if !NET6_0_OR_GREATER
+        //ERROR: Control cannot leave the body of an
+        //anonymous method or lambda expression
+       string[] args = {"/File", "fileThatMostCertainlyDoesNotExist"};
+       Func<string> f;
+       switch(args[0])
+       {
+        case "/File":
```
Comment (Keboo) [inline src/Chapter13/Table13.01.g.LambdaExpressionNotesAndExamples.cs]: nit: Typically case statements within a switch are indented.

File: src/Chapter13/Table13.01.i.LambdaExpressionNotesAndExamples.cs [hunk 13]
```diff
@@ -0,0 +1,24 @@
+﻿// Justification: Only snippets of source code shown for elucidation.
+#pragma warning disable CS0168 // Variable is declared but never used
+
+namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter13.Table13_01;
+
+public partial class LambdaExpressionNotesAndExamples
+{
+    // 9.
+    static public void UsingOutParameters()
```
Comment (Keboo) [inline src/Chapter13/Table13.01.i.LambdaExpressionNotesAndExamples.cs]:
```suggestion
    public static void UsingOutParameters()
```

File: src/Chapter13/Table13.01.j.LambdaExpressionNotesAndExamples.cs [hunk 14]
```diff
@@ -0,0 +1,24 @@
+﻿// Justification: Only snippets of source code shown for elucidation.
+#pragma warning disable CS0168 // Variable is declared but never used
+
+namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter13.Table13_01;
+
+public partial class LambdaExpressionNotesAndExamples
+{
+    // 10.
+    static public void CompilerWillNotDetectInLambdaAssignment()
```
Comment (Keboo) [inline src/Chapter13/Table13.01.j.LambdaExpressionNotesAndExamples.cs]:
```suggestion
    public static void CompilerWillNotDetectInLambdaAssignment()
```

File: src/Chapter13/Table13.01.j.LambdaExpressionNotesAndExamples.cs [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+    {
+//#if COMPILEERROR
+#if !NET6_0_OR_GREATER
+    int number;
+    Func<int, bool> isFortyTwo =
+        x => 42 == (number = x);
```
Comment (Keboo) [inline src/Chapter13/Table13.01.j.LambdaExpressionNotesAndExamples.cs]:
```suggestion
    Func<int, bool> isFortyTwo =
        x => 42 == x;
```

File: src/Chapter13/Table13.01.d.LambdaExpressionNotesAndExamples.cs (hunk 9 above)
Comment (BenjaminMichaelis) [reply src/Chapter13/Table13.01.d.LambdaExpressionNotesAndExamples.cs:11]: Keeping these here for continuity in the fact that these exist in the text.

**Review by Keboo — APPROVED** [review]

**Pull request merged**